src/TimerUnit.cpp

```diff
@@ -26,6 +26,13 @@
 #include "mudlet.h"
 #include "TTimer.h"
 
+TimerUnit::~TimerUnit()
+{
+    for (auto&& timer : qAsConst(mQTimerSet)) {
```

Comment:
Is that the right form? It seems to me that you are *move* constructing `timer` - but what is the `qAsConst(...)` doing - does it not prevent a copy being made of the items that are accessed in `mQTimerSet` so that suggests you are trying to both pull the items from the set and yet not modify the set...

:confused: 

TBH If it were me doing this I'd just use:
```cpp
    QMutableSetIterator<QTimer*> itPTimer(mQTimerSet);
    while (itPTimer.hasNext()) {
        delete itPTimer.next();
    }
```
fair enough that it is using a Qt Java style (?) iterator rather than a ranged `for` but it is clear how it works to me...

Reply:
Yeah that is the old-school iterator style. 

I am not constructing any timers, just looping the existing list.

Reply:
Have a look at https://en.cppreference.com/w/cpp/language/range-for for background

Reply:
:face_with_head_bandage: TBH I don't want to touch stuff like that - it may be nice and shiny and new-fangled but it confuses the hell out of me.... :older_man:

Reply:
Come on get in :)